examples/water_boiler/README.md

```diff
@@ -0,0 +1,42 @@
+# Water Boiler Example
+
+Simple simulation of a water boiler which can heat up water and where the heat dissipates slowly over time.
+
+## Installation
+
+Optionally, create a virtual environment for this example and activate it.
+
+```bash
+python -m venv water_boiler_venv # assuming Python 3
+. water_boiler_venv/bin/activate
+```
+
+If needed, install the [Matplotlib dependencies](https://stackoverflow.com/a/56673945/3767264) for your system.
```

Comment:
I don't have a lot of experience with troubleshooting Matplotlib installations myself, but this seems like a special case that most users probably don't have to deal with? I think the link could then be put directly in the troubleshooting section at the bottom to not confuse users into thinking they might need to do something here if they don't have to.

As an aside, could the problem in the question not be solved by running `sudo apt-get install python-tk` (or `sudo apt-get install python3-tk` if using python 3) instead of the proposed solution in the link? I think you use Ubuntu, right?

Reply:
> I think the link could then be put directly in the troubleshooting section

Yes.

I believe most Ubuntu users would have to do _something_. Possibly installing `python-tk` system-wide may be a way to solve the issue. Personally, and that's definitely a personal taste, I prefer recommending virtual-environment-based solution for that they are self-contained.

My own system-wide Python installation have tended to be short-lived, maybe because I don't use Python quite enough to remember what versions of what packages I installed for any given purpose. I like to rely on the cold efficiency of `pip install -r requirements.txt` :P Again, YMMV and I think it's you call, you're utlimately who is ensuring consitency in the documentation syles.

I'll move the link for now, please let me know what you prefer for the rest :slightly_smiling_face:

Reply:
My intention was to remove this line completely from the installation section and only mention matplotlib dependencies in the troubleshooting section, since I think many people wont have to worry about it. After this, I think this is ready for merging. It can always be updated later if we think of more things that should be included :slightly_smiling_face: 

I think managing python installations is not always easy, and I do tend to agree with you that keeping everything in a virtual environment that can easily be rebuilt using a simple `pip install -r requirements.txt` is usually the best way to go.

Reply:
Got it. I'll do this now : )



> It can always be updated later if we think of more things that should be included  :slightly_smiling_face:

Yes, that's always true! :+1: